platform-core/platform-telemetry/src/main/java/org/ekstep/telemetry/TelemetryGenerator.java

```diff
@@ -184,6 +188,8 @@ public static String audit(Map<String, String> context, List<String> props, Stri
 			edata.put("state", state);
 		if (StringUtils.isNotBlank(prevState))
 			edata.put("prevstate", prevState);
+		if(StringUtils.isNotBlank(context.get("duration")))
+			edata.put("duration",Long.valueOf(context.get("duration")));
```

Comment:
it is better to cast duration as Number and get longValue.

Reply:
as discussed, no change is required here.